Mathlib/Logic/Basic.lean

```diff
@@ -1280,4 +1280,13 @@ theorem ite_ite_comm (h : P → ¬Q) :
   dite_dite_comm P Q h
 #align ite_ite_comm ite_ite_comm
 
+variable {R : Prop}
+
+theorem ite_prop_iff_or : (if P then Q else R) ↔ (P ∧ Q ∨ ¬ P ∧ R) := by
+  by_cases p : P <;> simp [p]
+
+-- TODO make this a simp lemma in a future PR
+theorem ite_prop_iff_and : (if P then Q else R) ↔ ((P → Q) ∧ (¬ P → R)) := by
+  by_cases p : P <;> simp [p]
```

Comment:
Fancy adding @eric-wieser's `dite` versions here also? (from the thread)

Reply:
Will do, I'll make it a simp lemma at some later point, since it breaks quite some files.